external_stage.tf

```diff
@@ -8,7 +8,7 @@ resource "snowflake_stage" "this" {
 
   url                 = var.aws_s3_url
   storage_integration = var.storage_integration_name
-  file_format         = "TYPE = ${var.file_format} NULL_IF = []"
```

Comment:
this is a backwards incompatible change. I think it is fine as it is obvious, simple to fix and the version is still a `0.x`, however if you prefer to keep it backwards compatible, we could introduce a new `file_options` parameter that has a default of `NULL_IF = []`.

Reply:
👍🏼

Reply:
When you say backwards incompatible what do you mean? Could you leave a comment on a potential fix for older code? and what the version of the code that might break will look like?

Reply:
> When you say backwards incompatible what do you mean?

It means that if you currently have a `file_format = JSON` in your `*.auto.tfvars` then with this change, it would produce an invalid SQL statement, as it would resolve to:
```sql
JSON
```
instead of:
```sql
TYPE = JSON NULL_IF = []
```
inside the `snowflake_stage.this` resource

> Could you leave a comment on a potential fix for older code?

Yes, the fix would also be easy. You have to change your current:

```hcl
file_format = "JSON"
```
to
```
file_format = "TYPE = JSON NULL_IF = []"
```
when upgrading from `0.2.4` to `0.2.4+`

> and what the version of the code that might break will look like?

It would only break for people with `create_stage = true` and the internal `snowflake_stage` resource would not be created due to invalid parameters.

Reply:
@sfc-gh-pkommini let me know which way you prefer and I can adjust the pull request accordingly

Reply:
Sorry! The breaking change is okay. I meant could you leave a comment within the code to clarify the breaking change and the fix so that anyone who faces it might immediately notice the comment and make the fix within their code?